### ISSUE TITLE
Cryptic `No constructor for type Null$` error when parsing missing Option field with strict option parsing enabled

### ISSUE THREAD

**Issue by cstroe** [json4s/json4s#641]
### json4s version
3.6.7, master

### scala version
2.12.10

### jdk version
1.8.0_181

### Steps to reproduce

Declare the following case class:
```
case class OptionOfInt(opt: Option[Int])
```

Then try to extract it from the following parsed json:
```
{
  "opt": "not an int"
}
```

Using strict option parsing:
```
implicit val formats = new DefaultFormats {
  override val strictOptionParsing: Boolean = true
}
```

### What happens
An `org.json4s.package$MappingException` is thrown with the following exception message:
```
No constructor for type Null$, JString(not an int) 
```

### What should happen
An `org.json4s.package$MappingException` should be thrown with the following exception message:
```
No usable value for opt
Do not know how to convert JString(not an int) into int
```

**Comment by magnolia-k** [comment]
Looking at past commit logs, this commit is probably affecting.

https://github.com/json4s/json4s/commit/243da5b2e16e3a088dd05f91a2eb162290137ae5

I'm going to figure out how to fix this change without breaking it.

**Comment by magnolia-k** [comment]
fixed
https://github.com/json4s/json4s/pull/643

**Comment by seratch** [comment]
I can work on it.

**Comment by seratch** [comment]
Fixed by #642
